File: AtmLocator/app/src/main/java/piotrek/atmlocator/orm/Bank.java
package piotrek.atmlocator.orm;

import com.j256.ormlite.field.DatabaseField;
import com.j256.ormlite.table.DatabaseTable;

@DatabaseTable(tableName = "bank", daoClass = BankDao.class)
public class Bank {

    @DatabaseField(generatedId = true, columnName = Columns.ID)
    private int id;
    @DatabaseField(columnName = Columns.BANK_NAME, canBeNull = false, unique = true)
    private String name;
    @DatabaseField(columnName = Columns.PHONE)
    private String phone;

    public Bank(){ }

    @Override
    public String toString() {
        return name;
    }

    public Bank(String name, String phone){
        this();
        this.name = name;
        this.phone = phone;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public int getId() {

        return id;
    }

    public String getName() {
        return name;
    }

    public String getPhone() {
        return phone;
    }

    public class Columns {
        public static final String ID = "id";
        public static final String BANK_NAME = "bank_name";
        public static final String PHONE = "phone";
    }
}

File: AtmLocator/app/src/main/java/piotrek/atmlocator/orm/Bank.kt
package piotrek.atmlocator.orm

import com.j256.ormlite.field.DatabaseField
import com.j256.ormlite.table.DatabaseTable

@DatabaseTable(tableName = "bank", daoClass = BankDao::class)
class Bank() {

    @DatabaseField(generatedId = true, columnName = Columns.ID)
    var id: Int = 0
    @DatabaseField(columnName = Columns.BANK_NAME, canBeNull = false, unique = true)
    var name: String? = null
    @DatabaseField(columnName = Columns.PHONE)
    var phone: String? = null

    override fun toString(): String {
        return name ?: ""
    }

    constructor(name: String, phone: String) : this() {
        this.name = name
        this.phone = phone
    }

    object Columns {
        const val ID = "id"
        const val BANK_NAME = "bank_name"
        const val PHONE = "phone"
    }
}
